avm/ptn/finops-toolkit/finops-hub/main.bicep

```diff
@@ -0,0 +1,222 @@
+    'ftk-version': ftkVersion
+    'ftk-tool': 'FinOps hubs'
+  }
+)
+
+// Generate globally unique Data Factory name: 3-63 chars; letters, numbers, non-repeating dashes
+var uniqueSuffix = uniqueString(hubName, resourceGroup().id)
+var dataFactoryPrefix = '${replace(hubName, '_', '-')}-engine'
+var dataFactorySuffix = '-${uniqueSuffix}'
+var dataFactoryName = replace(
+  '${take(dataFactoryPrefix, 63 - length(dataFactorySuffix))}${dataFactorySuffix}',
+  '--',
+  '-'
+)
+
+// The last segment of the telemetryId is used to identify this module
+var telemetryId = '00f120b5-2007-6120-0000-40b000000000'
+
+//==============================================================================
+// Resources
+//==============================================================================
+
+//------------------------------------------------------------------------------
+// ADLSv2 storage account for staging and archive
+//------------------------------------------------------------------------------
+
+module storage 'modules/storage.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-storage'
+  params: {
+    storageAccountName: storageAccountName
+    sku: storageSku
+    location: location
+    tags: resourceTags
+    tagsByResource: tagsByResource
+    exportScopes: exportScopes
+    configContainer: configContainer
+    exportContainer: exportContainer
+    ingestionContainer: ingestionContainer
+    ftkVersion: ftkVersion
+  }
+}
+
+//------------------------------------------------------------------------------
+// Data Factory and pipelines
+//------------------------------------------------------------------------------
+resource dataFactory 'Microsoft.DataFactory/factories@2018-06-01' = {
+  name: dataFactoryName
+  location: location
+  tags: union(
+    resourceTags,
+    contains(tagsByResource, 'Microsoft.DataFactory/factories') ? tagsByResource['Microsoft.DataFactory/factories'] : {}
+  )
+  identity: { type: 'SystemAssigned' }
+  properties: any(
+    // Using any() to hide the error that gets surfaced because globalConfigurations is not in the ADF schema yet.
+    {
+      globalConfigurations: {
+        PipelineBillingEnabled: 'true'
+      }
+    }
+  )
+}
+module dataFactoryResources 'modules/dataFactory.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-dataFactoryResources'
+  params: {
+    dataFactoryName: dataFactoryName
+    convertToParquet: convertToParquet
+    keyVaultName: keyVault.outputs.name
+    storageAccountName: storage.outputs.name
+    exportContainerName: exportContainer
+    ingestionContainerName: ingestionContainer
+    location: location
+    tags: resourceTags
+    tagsByResource: tagsByResource
+  }
+}
+
+//------------------------------------------------------------------------------
+// Key Vault for storing secrets
+//------------------------------------------------------------------------------
+
+module keyVault 'modules/keyVault.bicep' = {
```

Comment:
My question here, Ideally we have patterns built on top of resource modules. would we not be better to leverage the AVM Modules instead of custom module calls for the top level resources. I understand we have a need to create resource before the nested loop creation can take place but I fear especially with WAF & Resiliency alignment we are losing the benefit of the AVM Atomic Modules by not using them

Reply:
@sebassem - Apologies this was supposed to be surrounding the data factory object not the KV

Reply:
If I understood correctly, this comment is about using the dataFactory resource and not the AVM module for it , correct? If yes, the problem is that there is a property that is not yet in the schema "PipelineBillingEnabled". If there is a way to add it in the AVM module, I can make the switch

Reply:
@clintgrove - Can you take a look at this so we can have this ptn module leverage your DF Module

Reply:
Commenting to bump this ask so we can finalize the module :)

Reply:
I agree with Seif, the use of any() is necessary here to add PipelineBillingEnabled and there is slim chance that this will be needed by more than even 5% of our users of the AVM module so its best to use the Factory resource call here instead of the AVM module call to Factory

Reply:
Happy to resolve this - thanks clint